File: Discord_android/Discord/app/src/main/java/com/daophilac/discord/JSONConverter.java
package com.daophilac.discord;

import com.daophilac.discord.models.Channel;
import com.daophilac.discord.models.Message;
import com.daophilac.discord.models.Server;
import com.daophilac.discord.models.User;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

public class JSONConverter {
    Gson gson;// = new Gson();
    GsonBuilder gson2;
    public JSONConverter(){
        gson = new Gson();
        gson2 = new GsonBuilder().setDateFormat("yyyy-MM-ddTHH:mm:ss");
        //gson.
    }
    public User toUser(String json){
        User user = gson.fromJson(json, User.class);
        return user;
    }
    public List<User> toListUser(String json){
        Type listUserType = new TypeToken<List<User>>(){}.getType();
        List<User> listUser = gson.fromJson(json, listUserType);
        return listUser;
    }
    public Server toServer(String json){
        Server server = gson.fromJson(json, Server.class);
        return server;
    }
    public List<Server> toListServer(String json){
        Type listServerType = new TypeToken<List<Server>>(){}.getType();
        List<Server> listServer = gson.fromJson(json, listServerType);
        return listServer;
    }
    public Channel toChannel(String json){
        Channel channel = gson.fromJson(json, Channel.class);
        return channel;
    }
    public List<Channel> toListChannel(String json){
        Type listChannelType = new TypeToken<List<Channel>>(){}.getType();
        List<Channel> listChannel = gson.fromJson(json, listChannelType);
        return listChannel;
    }
    public Message toMessage(String json){
        Message message = gson.fromJson(json, Message.class);
        return message;
    }
    public List<Message> toListMessage(String json){
        Type listMessageType = new TypeToken<List<Message>>(){}.getType();
        json = json.replace("T0", " 0");
        List<Message> listMessage = gson.fromJson(json, listMessageType);
        return listMessage;
    }
}

File: Discord_win/Discord_win/Tools/JSONConverter.cs
using Discord_win.Models;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Runtime.Serialization.Json;
using System.Text;
using System.Threading.Tasks;

namespace Discord_win.Tools {
    class JSONConverter {
        public User ToUser(string json) {
            User deserializedUser = new User();
            MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(json));
            DataContractJsonSerializer ser = new DataContractJsonSerializer(deserializedUser.GetType());
            deserializedUser = ser.ReadObject(ms) as User;
            ms.Close();
            return deserializedUser;
        }
        public List<User> ToListUser(string json) {
            List<User> deserializedListUser = new List<User>();
            MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(json));
            DataContractJsonSerializer ser = new DataContractJsonSerializer(deserializedListUser.GetType());
            deserializedListUser = ser.ReadObject(ms) as List<User>;
            ms.Close();
            return deserializedListUser;
        }
        public Server ToServer(string json) {
            Server deserializedServer = new Server();
            MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(json));
            DataContractJsonSerializer ser = new DataContractJsonSerializer(deserializedServer.GetType());
            deserializedServer = ser.ReadObject(ms) as Server;
            ms.Close();
            return deserializedServer;
        }
        public List<Server> ToListServer(string json) {
            List<Server> deserializedListServer = new List<Server>();
            MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(json));
            DataContractJsonSerializer ser = new DataContractJsonSerializer(deserializedListServer.GetType());
            deserializedListServer = ser.ReadObject(ms) as List<Server>;
            ms.Close();
            return deserializedListServer;
        }
        public Channel ToChannel(string json) {
            Channel deserializedChannel = new Channel();
            MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(json));
            DataContractJsonSerializer ser = new DataContractJsonSerializer(deserializedChannel.GetType());
            deserializedChannel = ser.ReadObject(ms) as Channel;
            ms.Close();
            return deserializedChannel;
        }
        public List<Channel> ToListChannel(string json) {
            List<Channel> deserializedListChannel = new List<Channel>();
            MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(json));
            DataContractJsonSerializer ser = new DataContractJsonSerializer(deserializedListChannel.GetType());
            deserializedListChannel = ser.ReadObject(ms) as List<Channel>;
            ms.Close();
            return deserializedListChannel;
        }
        public Message ToMessage(string json) {
            Message deserializedMessage = new Message();
            MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(json));
            DataContractJsonSerializer ser = new DataContractJsonSerializer(deserializedMessage.GetType());
            deserializedMessage = ser.ReadObject(ms) as Message;
            ms.Close();
            return deserializedMessage;
        }
        public List<Message> ToListMessage(string json) {
            List<Message> deserializedListMessage = new List<Message>();
            MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(json));
            DataContractJsonSerializer ser = new DataContractJsonSerializer(deserializedListMessage.GetType());
            deserializedListMessage = ser.ReadObject(ms) as List<Message>;
            ms.Close();
            return deserializedListMessage;
        }
    }
}
